semgrep_output_v1.atd

```diff
@@ -396,37 +394,60 @@ type svalue_value <python decorator="dataclass(frozen=True)"> = {
 (*****************************************************************************)
 (* Match dataflow trace *)
 (*****************************************************************************)
-
 (* EXPERIMENTAL *)
+
+(* It's easier to understand the dataflow trace data structures on a simple
+ * example. Here is one simple Python target file:
+ *
+ * 1:   def foo():
+ * 2:     return source()
+ * 3:   
+ * 4:   def bar(v):
+ * 5:     sink(v)
+ * 6:   
+ * 7:   x = foo()
+ * 8:   y = x
+ * 9:   bar(y)
+ *
+ * and here is roughly the generated match_dataflow_trace assuming
+ * a Semgrep rule where source() is a taint source and sink() the taint sink:
+ *
+ *  taint_source = CliCall("foo() @l7", [], CliLoc "source() @l2")
+ *  intermediate_vars = ["x", "y"]
+ *  taint_sink = CliCall("bar()" @l9, ["v"], CliLoc "sink(v) @l5")
+ *)
+
 type match_dataflow_trace <python decorator="dataclass(frozen=True)"> = {
   ?taint_source: match_call_trace option;
-  (* Intermediate variables which are involved in the dataflow. For taint, this
+  (* Intermediate variables which are involved in the dataflow. This
    * explains how the taint flows from the source to the sink. *)
   ?intermediate_vars: match_intermediate_var list option;
-  (* For now, omitting the taint sink, since it's redundant data except for
```

Comment:
@IagoAbal I've removed this comment as it appears to be incorrect.

Reply:
I'm not sure what the comment meant exactly, but inter-proc is now the default. I guess it meant that in intra-proc, the trace for the `taint_sink` will just be a `CliLoc` and it will be the same loc as the finding, so in that sense it is redundant.